tests/functional/test_collection_meta.py

```diff
@@ -0,0 +1,356 @@
+    removal:
+      major_version: 12
+      reason: guidelines-violation
+      reason_text: Refused to run CI.
+      new_name: namespace.name
+      discussion: https://forum.ansible.com/...
+      announce_version: 9.3.0
+  bad.foo9:
+    repository: https://github.com/ansible-collections/collection_template
+    removal:
+      major_version: 11
+      reason: other
+      reason_text: The collection wasn't cow friendly, so the Steering Committee decided to kick it out.
+      discussion: https://forum.ansible.com/...
+      announce_version: 9.3.0
+      redirect_replacement_major_version: 12
+  bad.foo10:
+    repository: https://github.com/ansible-collections/collection_template
+    removal:
+      major_version: 11
+      reason: considered-unmaintained
+      reason_text: Foo!
+      reason_other: This shouldn't be there!
+      discussion: https://forum.ansible.com/...
+  bad.foo11:
+    repository: https://github.com/ansible-collections/collection_template
+    removal:
+      major_version: 11
+      reason: considered-unmaintained
+      discussion: https://forum.ansible.com/...
+""",
+        [],
+        [
+            "collections -> bad.foo0 -> removal: Value error, new_name must be provided if reason is 'renamed'",
+            "collections -> bad.foo1 -> removal -> announce_version: Value error, must be a string or PypiVer object, got 42",
+            "collections -> bad.foo10 -> removal -> reason_other: Extra inputs are not permitted",
+            "collections -> bad.foo2 -> removal -> announce_version: Value error, must be a version with three release numbers (e.g. 1.2.3, 2.3.4a1), got '9.3'",
+            "collections -> bad.foo3 -> removal -> announce_version: Value error, must be a non-trivial string, got ''",
+            "collections -> bad.foo4 -> removal: Value error, major_version must not be TBD if reason is not 'renamed'",
+            "collections -> bad.foo5 -> removal: Value error, "
+            "redirect_replacement_major_version must be smaller than major_version",
+            "collections -> bad.foo6 -> removal: Value error, reason_text must not be provided if reason is not 'other', 'guidelines-violation'",
+            "collections -> bad.foo7 -> removal: Value error, reason_text must be provided if reason is 'other', 'guidelines-violation'",
+            "collections -> bad.foo8 -> removal: Value error, new_name must not be provided if reason is not 'renamed'",
+            "collections -> bad.foo9 -> removal: Value error, redirect_replacement_major_version must not be provided if reason is not 'renamed'",
+        ],
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "major_release, collection_metadata, all_collections, expected_errors",
+    LINT_COLLECTION_META_DATA,
+)
+def test_lint_collection_meta(
+    major_release: int,
+    collection_metadata: str,
+    all_collections: list[str],
+    expected_errors: list[str],
+    tmp_path,
```

Comment:
I usually annotate fixtures (and import Path from pathlib in this case), but that's up to you.

```suggestion
    tmp_path: Path,
```

Reply:
de87a96da267a1d2e4cc9a074a6596df30627e49